sound/soc/intel/boards/skl_hda_dsp_generic.c

```diff
@@ -206,7 +226,12 @@ static int skl_hda_audio_probe(struct platform_device *pdev)
 		hda_soc_card.components = hda_soc_components;
 	}
 
-	return devm_snd_soc_register_card(&pdev->dev, &hda_soc_card);
+	ret = devm_snd_soc_register_card(&pdev->dev, &hda_soc_card);
+
+	if (!ret)
+		skl_hda_set_ana_codec_autosuspend(&hda_soc_card);
+
```

Comment:
@jason77-wang  ,   This delay depends on platform, you could check https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c#L227 to apply it only to X1 or HP

Reply:
@RanderWang,
Thanks for your suggestion, I guess the root cause for this problem is there is a combo audio jack (headphone + mic) on the machine and the hardware design doesn't use MIC_DET for mic detecting, that makes the mic can't trigger the unsol event during runtime suspend.

And this hw design is very common on Lenovo and HP machines, for LENOVO machines,  We already have 8 models to support dmic and all of them have this issues.  I guess it is similar for HP machines. And this delay will not add much power consumption on the Dell machines, could we apply the delay to all machines instead of introducing a quirk table?

Reply:
@jason77-wang It depends on @ClarexZhou :).   And DELL adopt software solution for Jack, but HP & Lenovo use hardware detection, so this issue is not detected on DELL devices. The last test show regression issue with this PR. Hi @ClarexZhou how about mantis ?

Reply:
@Liviali155 has tested on Mantis, all work ok. No side effect on that.

Reply:
@plbossart, here's actually more likely explanation why the delay is needed:

 @jason77-wang wrote
> Thanks for your suggestion, I guess the root cause for this problem is there is a combo audio jack (headphone + mic) on the machine and the hardware design doesn't use MIC_DET for mic detecting, that makes the mic can't trigger the unsol event during runtime suspend.